optional_plugins/mail/avocado_mail_result/mail_result.py

```diff
@@ -0,0 +1,257 @@
+            )
+        msg["From"] = job.config.get("plugins.mail.sender")
+        msg["To"] = job.config.get("plugins.mail.recipient")
+        return msg
+
+    @staticmethod
+    def _send_email(smtp, sender, rcpt, msg):
+        try:
+            smtp.sendmail(sender, [rcpt], msg.as_string())
+            LOG_UI.info("EMAIL SENT TO: %s", rcpt)
+        except Exception as e:  # pylint: disable=W0703
+            LOG_UI.error(f"Failure to send email notification to {rcpt}: {e}")
+
+    @staticmethod
+    def _create_smtp_connection(server, port):
+        try:
+            smtp = smtplib.SMTP(server, port)
+            smtp.starttls()  # Enable TLS
+            return smtp
+        except Exception as e:  # pylint: disable=W0703
+            LOG_UI.error(f"Failed to establish SMTP connection to {server}:{port}: {e}")
+            return None
+
+    @staticmethod
+    def _read_results_file(results_path):
+        try:
+            with open(results_path, "r", encoding="utf-8") as file:
+                return json.load(file)
+        except FileNotFoundError:
+            LOG_UI.error("Test summary file not found at %s.", results_path)
+        except json.JSONDecodeError:
+            LOG_UI.error("Error decoding JSON from file %s.", results_path)
+        except Exception as e:
+            LOG_UI.error("Unexpected error while reading test summary: %s", str(e))
+        return None
+
+    @staticmethod
+    def _format_test_details(test, advanced=False):
+        if advanced:
+            details = [
+                f"<strong>Name:</strong> {test.get('name', '')}<br>",
+                f"<strong>Status:</strong> {test.get('status', '')}<br>",
+                f"<strong>Fail Reason:</strong> {test.get('fail_reason', '')}<br>",
+                f"<strong>Actual Time Start:</strong> {test.get('actual_time_start', '')}<br>",
+                f"<strong>Actual Time End:</strong> {test.get('actual_time_end', '')}<br>",
+                f"<strong>ID:</strong> {test.get('id', '')}<br>",
+                f"<strong>Log Directory:</strong> {test.get('logdir', '')}<br>",
+                f"<strong>Log File:</strong> {test.get('logfile', '')}<br>",
+                f"<strong>Time Elapsed:</strong> {test.get('time_elapsed', '')}<br>",
+                f"<strong>Time Start:</strong> {test.get('time_start', '')}<br>",
+                f"<strong>Time End:</strong> {test.get('time_end', '')}<br>",
+                f"<strong>Tags:</strong> {test.get('tags', '')}<br>",
+                f"<strong>Whiteboard:</strong> {test.get('whiteboard', '')}<br>",
+            ]
+        else:
+            details = [
+                f"<strong>Name:</strong> {test.get('name', '')}<br>",
+                f"<strong>Fail Reason:</strong> {test.get('fail_reason', '')}<br>",
+            ]
+        return "".join(details)
+
+    @staticmethod
+    def _generate_test_summary(data, detail_level):
+        test_summary = []
+
+        def format_test_details(test):
```

Comment:
Is there any reason for this inner method, instead of calling  `Mail._format_test_details(test, advanced=detail_level)` directly?